cases/performance/file_to_blackhole/notes.md

```diff
@@ -0,0 +1,144 @@
+include = ["logs/*.log"]
+
+[sinks.prometheus]
+type = "prometheus_exporter"
+inputs = ["internal"]
+address = "0.0.0.0:9001"
+
+[sinks.blackhole]
+type = "blackhole"
+inputs = ["file"]
+```
+
+New results, `bytes_per_second` held to 2.5 Mb/sec:
+
+| `duplicates` | `bytes_per_second ` |  `maximum_line_size_bytes` | `file_gen` bytes/sec throughput | file source `sum(rate(vector_processed_bytes_total[1m]))` | delta `file_gen`, `file` |
+| ------------ | ------------------- | ------------------- | ------------------------------- | --------------------------------------------------------- | ------------------------ |
+| 10           | 2.5 Mb/sec           | 10 Kb/sec           | 25 Mb/sec             | 25 Mb/sec            | 0 Kb/sec                |
+| 10           | 25 Mb/sec            | 10 Kb/sec           | 246 Mb/sec            | 219 Mb/sec           | 27 Mb/sec               |
+| 10           | 50 Mb/sec            | 10 Kb/sec           | 475 Mb/sec            | 221 Mb/sec           | 254 Mb/sec              |
+| 10           | 2.5 Mb/sec           | 100 Kb/sec          | 25 Mb/sec             | 25 Mb/sec            | 0 Kb/sec                |
+| 50           | 2.5 Mb/sec           | 10 Kb/sec           | 125 Mb/sec            | 125 Mb/sec           | 0 Mb/sec                |
+| 75           | 2.5 Mb/sec           | 10 Kb/sec           | 187 Mb/sec            | 187 Mb/sec           | 0 Mb/sec                |
+| 85           | 2.5 Mb/sec           | 10 Kb/sec           | 212 Mb/sec            | 212 Mb/sec           | 0 Mb/sec                |
+| 90           | 2.5 Mb/sec           | 10 Kb/sec           | 225 Mb/sec            | 222 Mb/sec           | 3 Mb/sec                |
+| 95           | 2.5 Mb/sec           | 10 Kb/sec           | 237 Mb/sec            | 216 Mb/sec           | 19 Mb/sec               |
+| 100          | 2.5 Mb/sec           | 10 Kb/sec           | 250 Mb/sec            | 221 Mb/sec           | 29 Mb/sec               |
+| 255          | 2.5 Mb/sec           | 10 Kb/sec           | 616 Mb/sec            | 222 Mb/sec           | 394 Mb/sec              |
+| 510          | 100 Kb/sec           | 1 Kb/sec            | 48 Mb/sec             | 40 Mb/sec            | 8 Mb/sec                |
+
+One thing stood out in the tests: if I allowed vector to develop a backlog and
+then stopped `file_gen` the throughput would gradually catch up, before
+inevitably falling to zero when all the bytes were processed. There are two
+dimensions that cause the file source to fall behind, based on the above table:
+total number of files and byte throughput of files. 510 files of relatively
+meager throughput -- 48 Mb/sec -- forced file source to fall behind by 8 Mb/sec,
+a relatively small number of files -- 10 -- producing at 50Mb/sec caused the
+file source to fall behind by 254 Mb/sec. The progenitor code for `file-source`
+was concerned with fairness and not throughput, which I believe shows today in
+the data.
```

Comment:
@lukesteensen new results with the config bug corrected. Thanks again for catching that.